libs/payments/currency/src/lib/currency.manager.ts

```diff
@@ -0,0 +1,56 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { Injectable } from '@nestjs/common';
+import {
+  CURRENCIES_TO_COUNTRIES,
+  SUPPORTED_PAYPAL_CURRENCIES,
+  VALID_COUNTRY_CODES,
+  VALID_CURRENCY_CODES,
+} from './currency.constants';
+import {
+  CurrencyCodeInvalidError,
+  CountryCodeInvalidError,
+  CurrencyCountryMismatchError,
+} from './currency.error';
+
+@Injectable()
+export class CurrencyManager {
+  constructor() {}
+
+  /**
+   * Verify that provided source country and plan currency are compatible with
+   * valid currencies and countries as listed in constants
+   *
+   * @param currency Currency of customer
+   * @param country Country of customer
+   * @returns True if currency is compatible with country, else throws error
+   */
+  assertCurrencyCompatibleWithCountry(
+    currency: string | null | undefined,
+    country: string | null | undefined
```

Comment:
This taking both null and undefined as arguments is somewhat odd - why does this need to take both? Shouldn't currency and country both be required? The first thing you do in the method is throw an error if either aren't provided, so why not require them outright?

Reply:
Will update.